### DISCUSSION TITLE
Questions about nuxt3 NuxtLoadingIndicator and NuxtPage components!

### DISCUSSION THREAD

**Discussion by xhc-code** [nuxt/nuxt#20277] [Questions]
I have a few questions, welcome to communicate and discuss together:
1. What are the functions of the name and route of the NuxtPage component? I don't understand the official explanation, especially the route attribute.
2. The NuxtLoadingIndicator component does not work, and no changes can be seen through the official examples.
3. Does Nuxt3 have instructions on the order of hook calls? I also didn't see the life cycle events of nuxt3 (I saw the life cycle diagram in nuxt2, I don't know if it is applicable to nuxt3).

https://nuxt.com/docs/api/components/nuxt-page
https://nuxt.com/docs/api/components/nuxt-loading-indicator#nuxtloading-indicator

**Comment by ChrisGV04** [comment]
Answering your question about `NuxtLoadingIndicator`... You need to set a color for the indicator to show up. For example:
```vue
<NuxtLoadingIndicator color="#10b981" />
```
This might be confusing because the docs don't explicitly show it in their example, but in the props section that's the only one that doesn't have a default. Maybe that will make it work for you.

---

And for your question about the lifecycle diagram, I believe it should be the same or very similar to the Vue one [the official docs](https://vuejs.org/guide/essentials/lifecycle.html#lifecycle-diagram)

**Reply by xhc-code** [reply]
I want to ask about the life cycle of nuxt3, because nuxt3 has a lot of custom hooks, I want to know when these hooks are called.

**Reply by ChrisGV04** [reply]
Here's the list of every custom Nuxt 3 hook:
https://nuxt.com/docs/api/advanced/hooks

Each one has a description of when they're called

**Reply by xhc-code** [reply]
ok

**Reply by xhc-code** [reply]
> Answering your question about `NuxtLoadingIndicator`... You need to set a color for the indicator to show up. For example:
> 
> ```
> <NuxtLoadingIndicator color="#10b981" />
> ```
> 
> This might be confusing because the docs don't explicitly show it in their example, but in the props section that's the only one that doesn't have a default. Maybe that will make it work for you.
> 
> And for your question about the lifecycle diagram, I believe it should be the same or very similar to the Vue one [the official docs](https://vuejs.org/guide/essentials/lifecycle.html#lifecycle-diagram)

![image](https://user-images.githubusercontent.com/28866025/232262877-cad8e0b0-1ade-44c1-8008-cc703327d116.png)
I looked at the document and found that adding colors still didn't work. I found that there were default values except for colors, so when I added colors and clicked on the route, there was no progress display information.

**Reply by ChrisGV04** [reply]
Hmmm did you try with a Hex color? In all of my projects I use a Hex color and I have never had any issues. And maybe use the latest Nuxt version just in case

**Reply by xhc-code** [reply]
> Hmmm did you try with a Hex color? In all of my projects I use a Hex color and I have never had any issues. And maybe use the latest Nuxt version just in case

I tried, and it didn't work. I just added the color attribute to the official demo, and tried both the color name and the hexadecimal color value. When I clicked on the route, there was no progress information on the page.
The following is the official link and official demo:
https://nuxt.com/docs/api/components/nuxt-loading-indicator#nuxtloading-indicator
https://stackblitz.com/github/nuxt/nuxt/tree/main/examples/routing/pages?file=app.vue&terminal=dev

**Comment by ChrisGV04** [comment]
Answering your question about `NuxtPage`...

The `name` prop is used for rendering multiple views (Called Named Views). There's an example on [the Vue Router docs](https://router.vuejs.org/guide/essentials/named-views.html) that explain that topic "Named Views".

I'm not sure about what does the `route` prop do, to be honest.

Edit: I believe that both of those properties are not very common to be used inside Nuxt 3, since they're populated automatically. Maybe they could be useful if you modify the router using a plugin.

**Reply by xhc-code** [reply]
Ok, thanks.
